code/weapon/weapons.cpp

```diff
@@ -2638,11 +2650,23 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		if (spawn_weap < MAX_SPAWN_TYPES_PER_WEAPON)
 		{
 			// Get delay out of the way before handling interval so we can do adjusted lifetime 
+			float delay = 0.f;
 			if (optional_string("+Delay:")) {
-				stuff_float(&wip->spawn_info[spawn_weap].spawn_interval_delay);
+				stuff_float(&delay);
+
+				// since curves are used for spawn rate now, we just parse the value into a curve of spawn rate over absolute age
+				// this will behave correctly even if it's being used together with other spawn rate curves
+				SCP_string curve_name = wip->name;
+				curve_name += "SpawnDelayCurve" + spawn_weap;
```

Comment:
A weapon can have multiple spawn weapons assigned with different delays, so you might want to include a number here or something.

Reply:
That's what the `+ spawn_weapon` is for. Should work, unless I'm missing something.

Reply:
Oh, my bad, I thought it was the weapon-to-be-spawned's name.